tests.py

```diff
@@ -0,0 +1,124 @@
+import copy
```

Comment:
How does one run the tests? Should we document this in the `README`?

Reply:
Also should we separate tests on a per-module basis, i.e. put the color and iterator tests into separate files? How is this normally done in python? I personally haven't written any extensive unit tests for python code in the past.

Reply:
I'll document this, but it is a runnable script in essence and you can run `setup.py test` to run them as well (although that has been deprecated…).

For organization we could either use [discovery](https://docs.python.org/3/library/unittest.html#test-discovery) or split them into multiple files and use `from test_module import *` in `tests.py` to magically pull in the other test cases.

These days, pytest seems much more popular, but I wanted to avoid an extra dependency if there's an unit testing library that works in the standard library.